api/src/services/collections.ts

```diff
@@ -98,6 +98,32 @@ export class CollectionsService {
 						};
 					}
 
+					const addFlag = (flag: string) => {
+						if (!field.meta) {
+							field.meta = {
+								special: [flag],
+							} as FieldMeta;
+						} else if (!field.meta.special) {
+							field.meta.special = [flag];
+						} else {
+							field.meta.special.push(flag);
+						}
+					};
```

Comment:
Lets move this into a shared util function (it's reused in fields)

Reply:
Done. Thanks! 🙌🏼